### PR TITLE
Notification Refactor

### PR DESCRIPTION
**Pull request by harsha1979**

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java [hunk 1]
```diff
@@ -1588,13 +1590,14 @@ protected boolean retryAuthenticationEnabled() {
      * @throws AuthenticationFailedException : In occasions of failing sending the email to the user.
      */
     protected void triggerEvent(String userName, String tenantDomain, String userStoreDomainName,
-                                String notificationEvent, String otpCode) throws AuthenticationFailedException {
+                                String notificationEvent, String otpCode, String email) throws AuthenticationFailedException {
```
Comment (ruwanta) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java:1593]: Can we rename the parameter to "String email" to "String sendToAddress"

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java [hunk 2]
```diff
@@ -1588,13 +1590,14 @@ protected boolean retryAuthenticationEnabled() {
      * @throws AuthenticationFailedException : In occasions of failing sending the email to the user.
      */
     protected void triggerEvent(String userName, String tenantDomain, String userStoreDomainName,
-                                String notificationEvent, String otpCode) throws AuthenticationFailedException {
```
Comment (ruwanta) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java:1591]: This is an API Change. better to deprecate this call, and have a new private method.

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java (hunk 1 above)
Comment (harsha1979) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java:1593]: that is a choice. you can change it.

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java (hunk 2 above)
Comment (harsha1979) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java:1591]: This existing API was introduced in last WUM update and that is why we thought to change it. If that is a problem, +1 to add another method.

**Pull request merged**